server/api.go

```diff
@@ -205,6 +248,12 @@ func (p *Plugin) handleDeletionUnlock(w http.ResponseWriter, r *http.Request) {
 }
 
 func (p *Plugin) handleGetConfig(w http.ResponseWriter, r *http.Request) {
+	userID := r.Header.Get("Mattermost-User-ID")
+	if userID == "" {
+		http.Error(w, "Not authorized", http.StatusUnauthorized)
+		return
+	}
```

Comment:
Could I pass an arbitrary string for this header's value and get through here? May be worth having a function `AuthenticateUser` that fetches the user based on the ID, and could even make sure it matches against the AllowedEmailDomains list? You could call this function [here](https://github.com/mattermost/mattermost-plugin-cloud/pull/146/files#diff-3fb3d4184590bc6f747661f3062676031e837b0dd5f2b18640f439be7a2235f6R50) so that it happens automatically for new endpoints